v03_pipeline/lib/tasks/update_variant_annotations_table_with_new_samples.py

```diff
@@ -246,38 +251,14 @@ def update_table(self, ht: hl.Table) -> hl.Table:
                 **get_fields(
                     ht,
                     self.dataset_type.sample_lookup_table_annotation_fns,
-                    **annotation_dependencies,
+                    **self.rdc_annotation_dependencies,
+                    **self.other_annotation_dependencies,
                     **self.param_kwargs,
                 ),
             )
 
         # 5) Fix up the globals.
-        ht = ht.annotate_globals(
-            paths=hl.Struct(),
-            versions=hl.Struct(),
-            enums=hl.Struct(),
-        )
-        for rdc in ReferenceDatasetCollection.for_reference_genome_dataset_type(
-            self.reference_genome,
-            self.dataset_type,
-        ):
-            rdc_ht = annotation_dependencies[f'{rdc.value}_ht']
-            rdc_globals = rdc_ht.index_globals()
-            ht = ht.annotate_globals(
-                paths=hl.Struct(
-                    **ht.globals.paths,
-                    **rdc_globals.paths,
-                ),
-                versions=hl.Struct(
-                    **ht.globals.versions,
-                    **rdc_globals.versions,
-                ),
-                enums=hl.Struct(
-                    **ht.globals.enums,
-                    **rdc_globals.enums,
-                ),
-            )
-        ht = annotate_enums(ht, self.reference_genome, self.dataset_type)
+        ht = self.fix_globals(ht, clear_globals=True)
```

Comment:
Could we get rid of the `clear_globals` arg if we managed the `rdc_globals` in the method but left `annotate_enums` here?

Reply:
```
        # 5) Fix up the globals.
        ht = ht.annotate_globals(
            paths=hl.Struct(),
            versions=hl.Struct(),
            enums=hl.Struct(),
        )
        ht = self.fix_globals(ht)
```
I realized this would probably work.

what's special about the enums in `annotate_enums`?

Reply:
Everything in `annotate_enums` is specific to fields that are annotated from the callset VCF and aren't coming from the reference data.  

It actually looks like we've introduced weirdness by slotting the clinvar enums into that file... which should be able to resolve by the upcoming work to make cached reference dataset query a task.